draft-ietf-teep-architecture.md

```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
-    a device before provisioning the TA to the device with a TEE.
-    Examples include the verification of the device 'root of trust'
-    and the type of TEE included in a device.
-
-  - A TEE in a device needs to determine whether a Device
-    Administrator (DA) or a Service Provider (SP) that wants to
-    manage an TA in the device is authorized to manage
-    applications in the TEE.
-
-  - Attestation must be able to ensure a TEE is genuine.
+Applications executing in a device are exposed to many different attacks 
+intended to compromise the execution of the application, or reveal the
+data upon which those applications are operating. These attacks increase
+with the number of other applications on the device, with such other
+applications coming from potentially untrustworthy sources. The 
+potential for attacks further increase with the complexity of features
+and applications on devices, and the unintented interactions among those
+features and applications. The danger of attacks on a system increases 
+as the sensitivity of the applications or data on the device increases.
+As an example, exposure of emails from a mail client is likely to be of 
+concern to its owner, but a compromise of a banking application raises 
+even greater concerns.
+
+The Trusted Execution Environment (TEE) concept is designed to execute
+applications in a protected environment that separates applications
+inside the TEE from the regular operating system and from other 
+applications on the device. This separation reduces the possibility
+of a successful attack on applications and the data contained inside the
+TEE. Typically, applications are chosen to execute inside a TEE because
+those applications perform security sensitive operations or operate on
+sensitive data. An application running inside a TEE is referred to as a 
+Trusted Applications (TA), while a normal application running in the 
+regular operating system is referred to as an Untrusted Application 
+(UA).
+
+The TEE uses hardware to enforce protections on the TA and its data, but
+also presents a more limited set of services to applications inside the
+TEE than is normally available to UA's running in the normal operating
+system.
+   
+But not all TEEs are the same, and different vendors may have different
+implementations of TEEs with different security properties, different
+features, and different control mechanisms to operate on TAs. Some
+vendors may themsleves market multiple different TEEs with different
+properties atuned to different markets. A device vendor may integrate
+one or more TEEs into their devices depending on market needs.
+
+To simplify the life of developers and service providers interacting
+with TAs in a TEE, an interoperable protocol for managing TAs running in
+different TEEs of various devices is needed. In this TEE ecosystem,
+there often arises a need for an external trusted party to verify the
+identity, claims, and rights of SPs, devices, and their TEEs. This
```

Comment:
"SPs" acronym is used before being defined

Reply:
To be updated.